components/case_macros/src/lib.rs

```diff
@@ -0,0 +1,65 @@
+// Copyright 2021 TiKV Project Authors. Licensed under Apache-2.0.
+
+//! Procedural macros used for case conversion.
+extern crate proc_macro;
+
+use proc_macro::{Group, Ident, Literal, TokenStream, TokenTree};
+
+fn transform_non_keyword_ident_to_string_liternal<Transform>(
+    ident: &Ident,
+    transform: Transform,
+) -> Literal
+where
+    Transform: FnOnce(String) -> String,
+{
+    Literal::string(&transform(ident.to_string()))
+}
```

Comment:
This function can be inlined.

Reply:
By "inline", I actually meant remove this function and directly use `Literal::string(&transform(ident.to_string()))` in `transform_idents_in_stream_to_string`.